### ISSUE TITLE
[Feature]: Add listener on Mini App close

### ISSUE THREAD

**Issue by sann05** [Telegram-Mini-Apps/telegram-apps#566]
### Is your feature request related to a problem? Please describe.

I need to handle the mini app close (send a request to save user progress), but I am struggling to find a way to do it.

Overall I found a workaround: 
1. I make the `BackButton` visible.
2. And adding event onClick: 
```
Telegram.WebApp.BackButton.onClick(function() {
    saveUserProgress();
    Telegram.WebApp.close();
});
```

### Describe the solution you'd like

I'd like to have something that natively allows to listen these events:
```
Telegram.WebApp.onEvent('mini_app_close', function() {
     saveUserProgress();
});
```

### Describe alternatives you've considered

I am mostly considering implementing this using vanilla js `telegram-web-app.js` library.

But I would also agree with using `telegram-apps/sdk` or alternative

### Additional context

 I am using the Unity WebGL application, which loads using vanilla JS,  so to not add complexity I would like to keep my stack the same.

**Comment by heyqbnk** [comment]
Will be implemented in https://github.com/Telegram-Mini-Apps/telegram-apps/issues/553

**Comment by veymay** [comment]
Hello friends, can you tell me how this works?

**Comment by sweeeeetch** [comment]
> Will be implemented in [#553](https://github.com/Telegram-Mini-Apps/telegram-apps/issues/553)

i assume it wasnt made, was it?

**Comment by heyqbnk** [comment]
> > Will be implemented in [#553](https://github.com/Telegram-Mini-Apps/telegram-apps/issues/553)
> 
> i assume it wasnt made, was it?

Nope. This should be implemented by Telegram devs. I implemented the `deactivated` event, but it doesn't work the way it is required in the issue.

https://github.com/Telegram-Mini-Apps/issues/issues/69
